### PR TITLE
Bump consul/proto-public and create mocks

### PR DESCRIPTION
**Pull request by jjti**
### Changes

This pulls in the latest public proto specifications from consul's proto-public package. We need this for a subsequent PR that references the newly created `TelemetryState` struct: https://github.com/hashicorp/consul-dataplane/pull/370

Bumping `proto-public` breaks the tests in `pkg/dns/dns_test.go` because the referenced mock is no longer exported. To get around that, while keeping the dns changes largely unchanged, a new mockery mock is created using the newly added make target below:

```make
# This generates mocks against public proto packages in consul. At the time of writing,
# only the dns and resource packages are used in consul-dataplane so only mocks for their
# interfaces are generated here.
.PHONY: mocks
mocks:
	for pkg in pbdns pbresource; do \
		mockery --srcpkg=github.com/hashicorp/consul/proto-public/$$pkg --output ./internal/mocks/$${pkg}mock --outpkg $${pkg}mock --case underscore --all; \
	done
```

The `pbresource` mocks are also added here as they are used in the next PR.

### Testing

These changes were pulled out of the follow-up PR where they were tested by building + deploying a custom build of consul-dataplane using these changes.

### CODE REVIEW

File: pkg/dns/dns_test.go [hunk 1]
```diff
@@ -40,7 +41,7 @@ func genRandomBytes(size int) (blk []byte) {
 }
 
 func (s *DNSTestSuite) Test_DisabledServer() {
-	mockedDNSConsulClient := pbdns.NewMockDNSServiceClient(s.T())
```
Comment (jjti) [inline pkg/dns/dns_test.go:43]: `NewMockDNSServiceClient` no longer exists in `github.com/hashicorp/consul/proto-public/pbdns`

**Comment by wilkermichael** [comment]
@jjti qq: is this meant to be supported in the next release (Consul 1.18/dataplane 1.14)?

**Comment by jjti** [comment]
> @jjti qq: is this meant to be supported in the next release (Consul 1.18/dataplane 1.14)?

Yes and that's correct

**Comment by jjti** [comment]
> @jjti qq: is this meant to be supported in the next release (Consul 1.18/dataplane 1.14)?

@wilkermichael yes that's correct

**Comment by wilkermichael** [comment]
Ok, we'll need to add a backport label, just waiting for it to be created. The release process is still ongoing for dataplane

File: .changelog/406.txt
Comment (DanStough) [inline .changelog/406.txt:1]: IMHO we don't need a changelog since this is an internal, intermediate change and you'll have a feature one for #370

**Review by DanStough — APPROVED** [review]
LGTM and thanks for this fix! See my comment about the changelog.

One rhetorical point that has nothing to do with this PR:  it's weird we got rid of the mocks to cut down on dependencies in `proto-public`, but we ended up pulling in a ton of K8s dependencies. 🤷

File: .changelog/406.txt
Comment (jjti) [reply .changelog/406.txt:1]: Makes sense! Removed the changelog

**Comment by jjti** [comment]
> > One rhetorical point that has nothing to do with this PR: it's weird we got rid of the mocks to cut down on dependencies in `proto-public`, but we ended up pulling in a ton of K8s dependencies. 🤷
> 
> I hear you @DanStough and that's something I'll call out here: it appears to increase the consul-dataplane binary size ~50% (19M > 29M):
> 
> ```shell
> # josh @ josh-XWGWHJ0JV2 in ~/GitHub/hashicorp/consul-dataplane on git:bump-protos
> $ make bin
> GOARCH=arm64 GOOS=linux CGO_ENABLED=0 go build -trimpath -buildvcs=false -ldflags="-X github.com/hashicorp/consul-dataplane/pkg/version.GitCommit=6ddd5cc" -o dist/linux/arm64/consul-dataplane ./cmd/consul-dataplane
> 
> # josh @ josh-XWGWHJ0JV2 in ~/GitHub/hashicorp/consul-dataplane on git:jjti/bump-protos o [16:26:52] 
> $ ll dist/linux/arm64/consul-dataplane                                    
> -rwxr-xr-x  1 josh  staff    29M Feb  2 16:22 dist/linux/arm64/consul-dataplane
> 
> # josh @ josh-XWGWHJ0JV2 in ~/GitHub/hashicorp/consul-dataplane on git:main o [16:33:17] 
> $ make bin
> GOARCH=arm64 GOOS=linux CGO_ENABLED=0 go build -trimpath -buildvcs=false -ldflags="-X github.com/hashicorp/consul-dataplane/pkg/version.GitCommit=6ddd5cc" -o dist/linux/arm64/consul-dataplane ./cmd/consul-dataplane
> 
> # josh @ josh-XWGWHJ0JV2 in ~/GitHub/hashicorp/consul-dataplane on git:main o [16:33:26] 
> $ ll dist/linux/arm64/consul-dataplane
> -rwxr-xr-x  1 josh  staff    19M Feb  2 16:33 dist/linux/arm64/consul-dataplane
> ```
> 
> Similar/same increase image for the image: 139MB to 150MB

There was a discussion about this in Slack. The gateway team will look at removing the generated code import of k8s to avoid this increase. Ie this increase in size, from this PR, is temporary until the next tagged release of `consul/proto-public` after which we'll be back to the size we'd been at before

**Pull request merged**